contracts/AelinUpFrontDeal.sol

```diff
@@ -0,0 +1,299 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "./AelinERC20.sol";
+import {AelinDeal} from "./AelinDeal.sol";
+import {AelinPool} from "./AelinPool.sol";
+import {IAelinPool} from "./interfaces/IAelinPool.sol";
+import {IAelinUpFrontDeal} from "./interfaces/IAelinUpFrontDeal.sol";
+import "./libraries/NftCheck.sol";
+import "./interfaces/ICryptoPunks.sol";
+
+contract AelinUpFrontDeal is AelinERC20, IAelinUpFrontDeal {
+    address constant CRYPTO_PUNKS = address(0xb47e3cd837dDF8e4c57F05d70Ab865de6e193BBB);
+
+    UpFrontPool public poolData;
+    UpFrontDeal public dealData;
+
+    mapping(address => uint256) public allowList;
+    mapping(address => IAelinPool.NftCollectionRules) public nftCollectionDetails;
+    mapping(address => mapping(address => bool)) public nftWalletUsedForPurchase;
+    mapping(address => mapping(uint256 => bool)) public nftId;
+    mapping(address => uint256) public amountAccepted;
+
+    uint256 public totalAmountAccepted;
+    uint256 public currentDealTokenTotal;
+    uint256 private poolStart;
+
+    bool private hasAllowList;
+    bool private hasNftList;
+    bool private dealDepositComplete;
+    bool public poolDepositComplete;
+
+    function initialize(
+        UpFrontPool calldata _poolData,
+        UpFrontDeal calldata _dealData,
+        uint256 _underlyingDealTokenAmount,
+        address _sponsor
+    ) external {
+        // pool initialization checks
+        require(_poolData.purchaseDuration >= 30 minutes && _poolData.purchaseDuration <= 30 days, "not within limit");
+        require(_poolData.sponsorFee <= 15e18, "exceeds max sponsor fee");
+        uint8 purchaseTokenDecimals = IERC20Decimals(_poolData.purchaseToken).decimals();
+        require(purchaseTokenDecimals <= 18, "purchase token not compatible");
+
+        // deal initialization checks
+        require(
+            _dealData.proRataRedemptionPeriod >= 30 minutes && _dealData.proRataRedemptionPeriod <= 30 days,
+            "30 mins - 30 days for prorata"
+        );
```

Comment:
Do we have a proRataRedemptionPeriod for up front deals? Since we are doing auto accept if the the total purchase token amount passes a minimum, there isn't a period after the deal is full to accept the deal. Since the deal is known up front, it is fair to say that purchasing is accepting the deal.

What I do think we need is a depositUnderlyingPeriod which starts after the minimum is passed where the holder must deposit the underlying, if they fail this period then all purchasers can withdraw.